src/context.ts

```diff
@@ -19,40 +29,23 @@ export function buildPath(path: ResponsePath | undefined) {
   return segments.reverse().join(".");
 }
 
-export interface SpanContext extends Object {
-  _spans: Map<string, Span>;
-  getSpanByPath(info: ResponsePath): Span | undefined;
-  addSpan(span: Span, info: GraphQLResolveInfo): void;
-  // Passed in from the outside context
-  requestSpan?: Span;
-}
-
-function isSpanContext(obj: any): obj is SpanContext {
-  return (
-    obj.getSpanByPath instanceof Function && obj.addSpan instanceof Function
-  );
-}
-
-export function requestIsInstanceContextRequest<CTX extends SpanContext>(
-  request: GraphQLRequestContext<CTX | Object>
-): request is GraphQLRequestContext<CTX> {
-  return isSpanContext(request.context);
-}
-
-// TODO: think about using symbols to hide these
-export function addContextHelpers(obj: any): SpanContext {
-  if (isSpanContext(obj)) {
-    return obj;
+export function addContextHelpers<Context extends ContextForApolloOpenTracing>(
+  contextValue: Context
+): WithRequiredSpanContext<Context> {
+  if (hasSpanContext(contextValue)) {
+    return contextValue;
   }
 
-  obj._spans = new Map<string, Span>();
-  obj.getSpanByPath = function (path: ResponsePath): Span | undefined {
-    return this._spans.get(buildPath(isArrayPath(path) ? path.prev : path));
-  };
-
-  obj.addSpan = function (span: Span, info: GraphQLResolveInfo): void {
-    this._spans.set(buildPath(info.path), span);
+  const spans = new Map<string, Span>();
```

Comment:
why was this attached to the context object? (closure here means it's no longer exposed)